src/NuGet.Core/NuGet.Versioning/VersionRangeFormatter.cs

```diff
@@ -65,18 +69,22 @@ public string Format(string format, object arg, IFormatProvider formatProvider)
                             }
                         }
 
-                        formatted = sb.ToString();
+                        string formatted = sb.ToString();
+                        return formatted;
                     }
                 }
             }
 
-            return formatted;
+#pragma warning disable CS8603 // Possible null reference return.
+            // bug? ICustomFormatter.Format doesn't appear to allow null return values
+            return null;
+#pragma warning restore CS8603 // Possible null reference return.
```

Comment:
Format will never really be null correct? Like we created a version object, so it will have a string representation?

Reply:
Similar to another comment, `format` can be null with this example: `string.Format(VersionRangeFormatter.Instance, "{0}", versionRange);`. If `{0}` is replaced with something like `{0:N}`, then `format == "N"`, but when the format string is `{0}`, then `format == null`.

However, this code has other codepaths where it will reach this null return value, like when the type passed into `arg` is neither castable to `IFormattable` or `VersionRange`.

My preference is to throw an `ArgumentException`. I don't suspect that any of NuGet's own code will ever hit the exception, but since this class is public, who knows if someone somewhere is doing something like that.